### ISSUE TITLE
DimensionMismatch("All data should contain same number of observations")

### ISSUE THREAD

**Issue by vinson2233** [FluxML/Flux.jl#1543]
Flux version : 0.11.6
Julia version : 1.5.4
OS : MacOS 11.2.3
Minimal code to reproduce the error : 
```
using CSV
using DataFrames
using Flux: Data.DataLoader

csv_file = CSV.File(download("https://gist.githubusercontent.com/michhar/2dfd2de0d4f8727f873422c5d959fff5/raw/fa71405126017e6a37bea592440b4bee94bf7b9e/titanic.csv")) # Read the data, use download() if the data from internet
df = DataFrame(csv_file) #Convert the csv=>Dataframe

X = df[:,Not(:Survived)]
y = df.Survived ;

train_data = DataLoader((convert(Matrix,X), reshape(y,length(y),1)), batchsize=1) # <- This line rises error
```

Error code
```
DimensionMismatch("All data should contain same number of observations")

Stacktrace:
 [1] _nobs at /Users/vinson.ciawandy/.julia/packages/Flux/goUGu/src/data/dataloader.jl:102 [inlined]
 [2] DataLoader(::Tuple{Array{Any,2},Array{Int64,2}}; batchsize::Int64, shuffle::Bool, partial::Bool) at /Users/vinson.ciawandy/.julia/packages/Flux/goUGu/src/data/dataloader.jl:71
 [3] DataLoader(::Array{Any,2}, ::Vararg{Any,N} where N; kws::Base.Iterators.Pairs{Symbol,Int64,Tuple{Symbol},NamedTuple{(:batchsize,),Tuple{Int64}}}) at ./deprecated.jl:72
 [4] top-level scope at In[5]:2
 [5] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1091
```

Why I think the error is weird :
`convert(Matrix,X)`  is 891×11 Array{Any,2}
`reshape(y,length(y),1)` is 891×1 Array{Int64,2}
Both of them are 2-D arrays with the same number of observations.

**Comment by DhairyaLGandhi** [comment]
Flux expects the last dimension to be the observations dim.
